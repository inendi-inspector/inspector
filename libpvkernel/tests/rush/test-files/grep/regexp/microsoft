Microsoft From Wikipedia, the free encyclopedia

Microsoft Corporation (NASDAQ: MSFT) is an American public multinational
corporation headquartered in Redmond, Washington, USA that develops,
manufactures, licenses, and supports a wide range of products and services
predominantly related to computing through its various product divisions.
Established on April 4, 1975 to develop and sell BASIC interpreters for the
Altair 8800, Microsoft rose to dominate the home computer operating system
market with MS-DOS in the mid-1980s, followed by the Microsoft Windows line of
operating systems. Microsoft would also come to dominate the office suite
market with Microsoft Office. The company has diversified in recent years into
the video game industry with the Xbox and its successor, the Xbox 360 as well
as into the consumer electronics and digital services market with Zune, MSN and
the Windows Phone OS. The ensuing rise of stock in the company's 1986 initial
public offering (IPO) made an estimated four billionaires and 12,000
millionaires from Microsoft employees. In May 2011, Microsoft Corporation
acquired Skype Communications for $8.5 billion dollars.[1]

Primarily in the 1990s, critics contend Microsoft used monopolistic business
practices and anti-competitive strategies including refusal to deal and tying,
put unreasonable restrictions in the use of its software, and used
misrepresentative marketing tactics; both the U.S. Department of Justice and
European Commission found the company in violation of antitrust laws. Known for
its interviewing process with obscure questions, various studies and ratings
were generally favorable to Microsoft's diversity within the company as well as
its overall environmental impact with the exception of the electronics portion
of the business.  Contents

	* 1 History
		  o 1.1 1984–1994: Windows and Office o 1.2 1995–2005: Internet and the
32-bit era o 1.3 2006–present: Vista and Cloud computing
	* 2 Product divisions
		  o 2.1 Windows & Windows Live Division, Server and Tools, Online
Services Division o 2.2 Business Division o 2.3 Entertainment and Devices
Division
	* 3 Culture 4 Corporate affairs
		  o 4.1 Financial o 4.2 Environment o 4.3 Marketing
	* 5 See also 6 References 7 External links

History Main articles: History of Microsoft and History of Microsoft Windows
Paul Allen and Bill Gates (respectively) on October 19, 1981 in a sea of PCs
after signing a pivotal contract. IBM called Microsoft in July 1980 inquiring
about programming languages for its upcoming PC line;[2] after failed
negotiations with another company, IBM gave Microsoft a contract to develop the
OS for the new line of PCs.[3]

Paul Allen and Bill Gates, childhood friends with a passion in computer
programming, were seeking to make a successful business utilizing their shared
skills. The January 1975 issue of Popular Electronics featured Micro
Instrumentation and Telemetry Systems's (MITS) Altair 8800 microcomputer. Allen
noticed that they could program a BASIC interpreter for the device; after a
call from Gates claiming to have a working interpreter, MITS requested a
demonstration. Since they didn't actually have one, Allen worked on a simulator
for the Altair while Gates developed the interpreter. Although they developed
the interpreter on a simulator and not the actual device, the interpreter
worked flawlessly when they demonstrated the interpreter to MITS in
Albuquerque, New Mexico in March 1975; MITS agreed to distribute it, marketing
it as Altair BASIC.[4] They officially established Microsoft on April 4, 1975,
with Gates as the CEO.[5] Allen came up with the original name of "Micro-Soft,"
as recounted in a 1995 Fortune magazine article. In August 1977 the company
formed an agreement with ASCII Magazine in Japan, resulting in its first
international office, "ASCII Microsoft".[6] The company moved to a new home in
Bellevue, Washington in January 1979.[5]

Microsoft entered the OS business in 1980 with its own version of Unix, called
Xenix.[7] However, it was DOS (Disk Operating System) that solidified the
company's dominance. After negotiations with Digital Research failed, IBM
awarded a contract to Microsoft in November 1980 to provide a version of the
CP/M OS, which was set to be used in the upcoming IBM Personal Computer (IBM
PC).[8] For this deal, Microsoft purchased a CP/M clone called 86-DOS from
Seattle Computer Products, branding it as MS-DOS, which IBM rebranded to
PC-DOS. Following the release of the IBM PC in August 1981, Microsoft retained
ownership of MS-DOS. Since IBM copyrighted the IBM PC BIOS, other companies had
to reverse engineer it in order for non-IBM hardware to run as IBM PC
compatibles, but no such restriction applied to the operating systems. Due to
various factors, such as MS-DOS's available software selection[citation
needed], Microsoft eventually became the leading PC OS vendor.[3][9] The
company expanded into new markets with the release of the Microsoft Mouse in
1983, as well as a publishing division named Microsoft Press.[10] Paul Allen
resigned from Microsoft in February after developing Hodgkin's disease.[11]
1984–1994: Windows and Office

While jointly developing a new OS with IBM in 1984, OS/2, Microsoft released
Microsoft Windows, a graphical extension for MS-DOS, on November 20.[12]
Microsoft moved its headquarters to Redmond on February 26, 1986, and on March
13 the company went public;[13] the ensuing rise in the stock would make an
estimated four billionaires and 12,000 millionaires from Microsoft
employees.[14] Due to the partnership with IBM, in 1990 the Federal Trade
Commission set its eye on Microsoft for possible collusion; it marked the
beginning of over a decade of legal clashes with the U.S. Government.[15]
Microsoft announced the release of its version of OS/2 to original equipment
manufacturers (OEMs) on April 2, 1987;[16] meanwhile, the company was at work
on a 32-bit OS, Microsoft Windows NT, using ideas from OS/2; it shipped on July
21, 1993 with a new modular kernel and the Win32 application programming
interface (API), making porting from 16-bit (MS-DOS-based) Windows easier. Once
Microsoft informed IBM of NT, the OS/2 partnership deteriorated.[17]

Microsoft introduced its office suite, Microsoft Office, in 1990. The software
bundled separate office productivity applications, such as Microsoft Word and
Microsoft Excel.[18] On May 22 Microsoft launched Windows 3.0 with a
streamlined user interface graphics and improved protected mode capability for
the Intel 386 processor.[19] Both Office and Windows became dominant in their
respective areas.[20][21] Novell, a Word competitor from 1984–1986, filed a
lawsuit years later claiming that Microsoft left part of its APIs undocumented
in order to gain a competitive advantage.[22]

On July 27, 1994, the U.S. Department of Justice, Antitrust Division filed a
Competitive Impact Statement that said, in part: "Beginning in 1988, and
continuing until July 15, 1994, Microsoft induced many OEMs to execute
anticompetitive "per processor" licenses. Under a per processor license, an OEM
pays Microsoft a royalty for each computer it sells containing a particular
microprocessor, whether the OEM sells the computer with a Microsoft operating
system or a non-Microsoft operating system. In effect, the royalty payment to
Microsoft when no Microsoft product is being used acts as a penalty, or tax, on
the OEM's use of a competing PC operating system. Since 1988, Microsoft's use
of per processor licenses has increased."[23] 1995–2005: Internet and the
32-bit era Bill Gates giving his deposition in 1998 for the United States v.
Microsoft trial. Once the U.S. Department of Justice 1993 took over from the
Federal Trade Commission, a protracted legal wrangling between Microsoft and
the department ensued, resulting in various settlements and possible blocked
mergers. Microsoft would point to companies such as AOL-Time Warner in its
defense.[15]

Following Bill Gates's internal "Internet Tidal Wave memo" on May 26, 1995
Microsoft began to redefine its offerings and expand its product line into
computer networking and the World Wide Web.[24] The company released Windows 95
on August 24, 1995, featuring pre-emptive multitasking, a completely new user
interface with a novel start button, and 32-bit compatibility; similar to NT,
it provided the Win32 API.[25][26] Windows 95 came bundled with the online
service MSN, and for OEMs Internet Explorer, a web browser. Internet Explorer
was not bundled with the retail Windows 95 boxes because the boxes were printed
before the team finished the web browser, and instead was included in the
Windows 95 Plus! pack.[27] Branching out into new markets in 1996, Microsoft
and NBC Universal created a new 24/7 cable news station, MSNBC.[28] Microsoft
created Windows CE 1.0, a new OS designed for devices with low memory and other
constraints, such as personal digital assistants.[29] In October 1997, the
Justice Department filed a motion in the Federal District Court, stating that
Microsoft violated an agreement signed in 1994 and asked the court to stop the
bundling of Internet Explorer with Windows.[30]

Bill Gates handed over the CEO position on January 13, 2000 to Steve Ballmer,
an old college friend of Gates and employee of the company since 1980, creating
a new position for himself as Chief Software Architect.[5][31] Various
companies including Microsoft formed the Trusted Computing Platform Alliance in
October 1999 to, among other things, increase security and protect intellectual
property through identifying changes in hardware and software. Critics decry
the alliance as a way to enforce indiscriminate restrictions over how consumers
use software, and over how computers behave, a form of digital rights
management; for example the scenario where a computer is not only secured for
its owner, but also secured against its owner as well.[32][33] On April 3,
2000, a judgment was handed down in the case of United States v. Microsoft,[34]
calling the company an "abusive monopoly";[35] it settled with the U.S.
Department of Justice in 2004.[13] On October 25, 2001 Microsoft released
Windows XP, unifying the mainstream and NT lines under the NT codebase.[36] The
company released the Xbox later that year, entering the game console market
dominated by Sony and Nintendo.[37] In March 2004 the European Union brought
antitrust legal action against the company, citing it abused its dominance with
the Windows OS, resulting in a judgment of €497 million ($613 million) and to
produce new versions of Windows XP without Windows Media Player, Windows XP
Home Edition N and Windows XP Professional N.[38][39] 2006–present: Vista and
Cloud computing CEO Steve Ballmer at the MIX event in 2008. In an interview
about his management style in 2005, he mentioned that his first priority was to
get the people he delegates to in order. Ballmer also emphasized the need to
continue pursuing new technologies even if initial attempts fail, citing the
original attempts with Windows as an example.[40]

Released in January 2007, the next version of Windows, Windows Vista, focused
on features, security, and a redesigned user interface dubbed Aero.[41][42]
Microsoft Office 2007, released at the same time, featured a "Ribbon" user
interface which was a significant departure from its predecessors. Relatively
strong sales of both titles helped to produce a record profit in 2007.[43] The
European Union imposed another fine of €899 million ($1.4 billion) for
Microsoft's lack of compliance with the March 2004 judgment on February 27,
2008, saying that the company charged rivals unreasonable prices for key
information about its workgroup and backoffice servers. Microsoft stated that
it was in compliance and that "these fines are about the past issues that have
been resolved".[44]

Bill Gates retired from his role as Chief Software Architect on June 27, 2008
while retaining other positions related to the company in addition to being an
advisor for the company on key projects.[45] Azure Services Platform, the
company's entry into the cloud computing market for Windows, launched on
October 27, 2008.[46] On February 12, 2009, Microsoft announced its intent to
open a chain of Microsoft-branded retail stores, and on October 22, 2009 the
first retail Microsoft Store opened in Scottsdale, Arizona; the same day the
first store opened Windows 7 was officially released to the public. Windows 7's
focus was on refining Vista with ease of use features and performance
enhancements, rather than a large reworking of Windows.[47][48][49]

Microsoft is a founding member of the Open Networking Foundation started on
March 23, 2011. Other founding companies include Google, HP Networking, Yahoo,
Verizon, Deutsche Telekom and 17 other companies. The nonprofit organization is
focused on providing support for a new cloud computing initiative called
Software-Defined Networking.[50] The initiative is meant to speed innovation
through simple software changes in telecommunications networks, wireless
networks, data centers and other networking areas.[51] Product divisions Main
article: Microsoft Product Divisions

For the 2010 fiscal year, Microsoft had five product divisions: Windows &
Windows Live Division, Server and Tools, Online Services Division, Microsoft
Business Division, and Entertainment and Devices Division.  Windows & Windows
Live Division, Server and Tools, Online Services Division

The company's Client division produces the flagship Windows OS line such as
Windows 7; it also produces the Windows Live family of products and services.
Server and Tools produces the server versions of Windows, such as Windows
Server 2008 R2 as well as a set of development tools called Microsoft Visual
Studio, Microsoft Silverlight, a web application framework, and Systems
Management Server, a collection of tools providing remote-control abilities,
patch management, software distribution and a hardware/software inventory.
Other server products include: Microsoft SQL Server, a relational database
management system, Microsoft Exchange Server, for certain business-oriented
e-mail and scheduling features, Small Business Server, for messaging and other
small business-oriented features; and Microsoft BizTalk Server, for business
process management.

Microsoft provides IT consulting ("Microsoft Consulting Services") and produces
a set of certification programs handled by the Server and Tools division
designed to recognize individuals who have a minimal set of proficiencies in a
specific role; this includes developers ("Microsoft Certified Solution
Developer"), system/network analysts ("Microsoft Certified Systems Engineer"),
trainers ("Microsoft Certified Trainers") and administrators ("Microsoft
Certified Systems Administrator" and "Microsoft Certified Database
Administrator"). Microsoft Press, which publishes books, is also managed by the
division. The Online Services Business division handles the online service MSN
and the search engine Bing. As of December 2009, the company also possesses an
18% ownership of the cable news channel MSNBC without any editorial control;
however, the division develops the channel's website, msnbc.com, in a joint
venture with the channel's co-owner, NBC Universal.[52] Business Division Front
entrance to building 17 on the main campus of the company's Redmond campus.

The Microsoft Business Division produces Microsoft Office including Microsoft
Office 2010, the company's line of office software. The software product
includes Word (a word processor), Access (a relational database program), Excel
(a spreadsheet program), Outlook (Groupware, frequently used with Exchange
Server), PowerPoint (presentation software), and Publisher (desktop publishing
software). A number of other products were added later with the release of
Office 2003 including Visio, Project, MapPoint, InfoPath and OneNote. The
division also develops enterprise resource planning (ERP) software for
companies under the Microsoft Dynamics brand. These include: Microsoft Dynamics
AX, Microsoft Dynamics NAV, Microsoft Dynamics GP, and Microsoft Dynamics SL.
They are targeted at varying company types and countries, and limited to
organizations with under 7,500 employees.[53] Also included under the Dynamics
brand is the customer relationship management software Microsoft Dynamics CRM,
part of the Azure Services Platform.  Entertainment and Devices Division

The Entertainment and Devices Division produces the Windows CE OS for embedded
systems and Windows Phone 7 for smartphones.[54] Microsoft initially entered
the mobile market through Windows CE for handheld devices, eventually
developing into the Windows Mobile OS and now, Windows Phone 7. Windows CE is
designed for devices where the OS may not directly be visible to the end user,
in particular, appliances and cars. The division also produces computer games
that run on Windows PCs and other systems including titles such as Age of
Empires, Halo and the Microsoft Flight Simulator series, and houses the
Macintosh Business Unit which produces Mac OS software including Microsoft
Office 2008 for Mac. Microsoft's Entertainment and Devices Division designs,
markets, and manufactures consumer electronics including the Xbox 360 game
console, the handheld Zune media player, and the television-based Internet
appliance MSN TV. Microsoft also markets personal computer hardware including
mice, keyboards, and various game controllers such as joysticks and gamepads.
Culture

Technical reference for developers and articles for various Microsoft magazines
such as Microsoft Systems Journal (or MSJ) are available through the Microsoft
Developer Network (MSDN). MSDN also offers subscriptions for companies and
individuals, and the more expensive subscriptions usually offer access to
pre-release beta versions of Microsoft software.[55][56] In April 2004
Microsoft launched a community site for developers and users, titled Channel9,
that provides a wiki and an Internet forum.[57] Another community site that
provides daily videocasts and other services, On10.net, launched on March 3,
2006.[58] Free technical support is traditionally provided through online
Usenet newsgroups, and CompuServe in the past, monitored by Microsoft
employees; there can be several newsgroups for a single product. Helpful people
can be elected by peers or Microsoft employees for Microsoft Most Valuable
Professional (MVP) status, which entitles them to a sort of special social
status and possibilities for awards and other benefits.[59]

Noted for its internal lexicon, the expression "eating our own dog food" is
used to describe the policy of using prerelease and beta versions of products
inside Microsoft in an effort to test them in "real-world" situations.[60] This
is usually shortened to just "dog food" and is used as noun, verb, and
adjective. Another bit of jargon, FYIFV or FYIV ("Fuck You, I'm [Fully]
Vested"), is used by an employee to indicate they are financially independent
and can avoid work anytime they wish.[61] The company is also known for its
hiring process, mimicked in other organizations and dubbed the "Microsoft
interview", which is notorious for off-the-wall questions such as "Why is a
manhole cover round?".[62]

Microsoft is an outspoken opponent of the cap on H1B visas, which allow
companies in the U.S. to employ certain foreign workers. Bill Gates claims the
cap on H1B visas make it difficult to hire employees for the company, stating
"I'd certainly get rid of the H1B cap" in 2005.[63] Critics of H1B visas argue
that increasing (or eliminating) the cap only takes jobs away from U.S.
citizens due to H1B workers working for lower salaries.[64] The Human Rights
Campaign Corporate Equality Index, a report of how progressive the organization
deems company policies towards LGBT (lesbian, gay, bisexual and transsexual)
employees, rated Microsoft as 87% from 2002 to 2004 and as 100% from 2005 to
2010 after they allowed gender expression.[65] Corporate affairs

The company is run by a board of directors made up of mostly company outsiders,
as is customary for publicly traded companies. Members of the board of
directors as of June 2010 are: Steve Ballmer, Dina Dublon, Bill Gates
(chairman), Raymond Gilmartin, Reed Hastings, Maria Klawe, David Marquardt,
Charles Noski, and Helmut Panke.[66] Board members are elected every year at
the annual shareholders' meeting using a majority vote system. There are five
committees within the board which oversee more specific matters. These
committees include the Audit Committee, which handles accounting issues with
the company including auditing and reporting; the Compensation Committee, which
approves compensation for the CEO and other employees of the company; the
Finance Committee, which handles financial matters such as proposing mergers
and acquisitions; the Governance and Nominating Committee, which handles
various corporate matters including nomination of the board; and the Antitrust
Compliance Committee, which attempts to prevent company practices from
violating antitrust laws.[67] Five year history graph of (NASDAQ:MSFT) stock on
September 29, 2009.[68]

When Microsoft went public and launched its initial public offering (IPO) in
1986, the opening stock price was $21; after the trading day, the price closed
at $27.75. As of July 2010, with the company's nine stock splits, any IPO
shares would be multiplied by 288; if one was to buy the IPO today given the
splits and other factors, it would cost about 9 cents.[69][70][71] The stock
price peaked in 1999 at around $119 ($60.928 adjusting for splits).[72] The
company began to offer a dividend on January 16, 2003, starting at eight cents
per share for the fiscal year followed by a dividend of sixteen cents per share
the subsequent year, switching from yearly to quarterly dividends in 2005 with
eight cents a share per quarter and a special one-time payout of three dollars
per share for the second quarter of the fiscal year.[72][73] Though the company
had subsequent increases in dividend payouts, the price of Microsoft's stock
remained steady for years.[73][74]

One of Microsoft's business tactics, described by an executive as "embrace,
extend and extinguish," initially embraces a competing standard or product,
then extends it to produce their own version which is then incompatible with
the standard, which in time extinguishes competition that does not or cannot
use Microsoft's new version.[75] Various companies and governments sue
Microsoft over this set of tactics, resulting in billions of dollars in rulings
against the company.[76][34][39] Microsoft claims that the original strategy is
not anti-competitive, but rather an exercise of its discretion to implement
features it believes customers want.[77] Financial

Standard and Poor's and Moody's have both given a AAA rating to Microsoft,
whose assets were valued at $41 billion as compared to only $8.5 billion in
unsecured debt. Consequently, in February 2011 Microsoft released a corporate
bond amounting to $2.25 billion with relatively low borrowing rates compared to
government bonds.[78]

For the first time in 20 years Apple Inc. surpassed Microsoft in Q1 2011
quarterly profits and revenues due to a slowdown in PC sales and continuing
huge losses in Microsoft's Online Services Division (which contains its search
engine Bing). Microsoft profits were $5.2 billion, while Apple Inc. profits
were $6 billion, on revenues of $14.5 billion and $24.7 billion
respectively.[79]

Microsoft's Online Services Division has been continuously loss-making since
2006 and in Q1 2011 it lost $726 million. This follows a loss of $2.5 billion
for the year 2010.[80] Environment

Microsoft is ranked on the 17th place in Greenpeace’s Guide to Greener
Electronics that ranks 18 electronics manufacturers according to their policies
on toxic chemicals, recycling and climate change.[81] Microsoft’s timeline for
phasing out BFRs and phthalates in all products is 2012 but its commitment to
phasing out PVC is not clear. As yet (January 2011) it has no products that are
completely free from PVC and BFRs.[82]

Microsoft's main U.S. campus received a silver certification from the
Leadership in Energy and Environmental Design (LEED) program in 2008, and it
installed over 2,000 solar panels on top of its buildings in its Silicon Valley
campus, generating approximately 15 percent of the total energy needed by the
facilities in April 2005.[83]

Microsoft makes use of alternative forms of transit. It created one of the
worlds largest private bus systems, the "Connector", to transport people from
outside the company; for on-campus transportation, the "Shuttle Connect" uses a
large fleet of hybrid cars to save fuel. The company also subsidises regional
public transport as an incentive.[83][84] In February 2010 however, Microsoft
took a stance against adding additional public transport and high-occupancy
vehicle (HOV) lanes to a bridge connecting Redmond to Seattle; the company did
not want to delay the construction any further.[85] Marketing

In 2004, Microsoft commissioned research firms to do independent studies
comparing the total cost of ownership (TCO) of Windows Server 2003 to Linux;
the firms concluded that companies found Windows easier to administrate than
Linux, thus those using Windows would administrate faster resulting in lower
costs for their company (i.e. lower TCO).[86] This spurred a wave of related
studies; a study by the Yankee Group concluded that upgrading from one version
of Windows Server to another costs a fraction of the switching costs from
Windows Server to Linux, although companies surveyed noted the increased
security and reliability of Linux servers and concern about being locked into
using Microsoft products.[87] Another study, released by the OSDL, claimed that
the Microsoft studies were "simply outdated and one-sided" and their survey
concluded that the TCO of Linux was lower due to Linux administrators managing
more servers on average and other reasons.[88]

As part of the "Get the Facts" campaign Microsoft highlighted the .NET trading
platform that it had developed in partnership with Accenture for the London
Stock Exchange, claiming that it provided "five nines" reliability. After
suffering extended downtime and unreliability[89][90] the LSE announced in 2009
that it was planning to drop its Microsoft solution and switch to a Linux based
one in 2010.[91][92]

Microsoft adopted the so-called "Pac-Man Logo", designed by Scott Baker, in
1987. Baker stated "The new logo, in Helvetica italic typeface, has a slash
between the o and s  to emphasize the "soft" part of the name and convey motion
and speed."[93] Dave Norris ran an internal joke campaign to save the old logo,
which was green, in all uppercase, and featured a fanciful letter O, nicknamed
the blibbet, but it was discarded.[94] Microsoft's logo with the "Your
potential. Our passion." tagline below the main corporate name, is based on a
slogan Microsoft used in 2008. In 2002, the company started using the logo in
the United States and eventually started a TV campaign with the slogan, changed
from the previous tagline of "Where do you want to go today?".[95][96][97]
During the private MGX (Microsoft Global Exchange) conference in 2010,
Microsoft unveiled the company's next tagline, "Be What's Next.", as well as a
new logo scheduled for use sometime in the future.[98] 
