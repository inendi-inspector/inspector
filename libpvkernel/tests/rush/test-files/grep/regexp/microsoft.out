October 1999 to, among other things, increase security and protect intellectual
on features, security, and a redesigned user interface dubbed Aero.[41][42]
security and reliability of Linux servers and concern about being locked into
3/413 elements are valid.
